### ISSUE TITLE
Build failure: python3.12-django-4.2.11

### ISSUE THREAD

**Issue by johnhamelink** [NixOS/nixpkgs#304339]
### Steps To Reproduce

1. build python3.12-django-4.2.11

### Build log

```
https://gist.github.com/johnhamelink/0639b80deddb48c174bb4dca3a0642fe

...

error: 1 dependencies of derivation '/nix/store/s1gxk48i45i1ylmd7ikc5pmbb16mp56b-python3.12-aws-xray-sdk-2.13.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/m20nbz110p56bijvwr1hnvdx0636kjaz-python3.12-factory-boy-3.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vm0qzf5lsrr36cpfky6wc1pnxc0g4mxq-python3.12-moto-5.0.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pq3kz55zglf1xslqb1gij0nw52bl1drq-python3.12-pytest-randomly-3.13.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/22jh889q66pa7zs8w9b11g3ibzkrkwbc-python3.12-httplib2-0.22.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2zclmn9y5pg0araxar5q6aqkh9dsi80c-python3.12-pycognito-2024.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/66yg39vs0kyqprwvr58a269v0ixb38sx-unit-home-assistant.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mgg5y6qmn9v3alhfwf0d370hzl2j18ab-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ncjkf4wxp6nrr4rg08mh48xzdh543dwp-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/m07p48yy42i5xf4ngi88bliz8bf7n8df-nixos-system-sun-24.05.20240415.4d2386d.drv' failed to build
```

### Notify maintainers

@mweinelt 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
john@sun:~/ > nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.25, NixOS, 24.05 (Uakari), 24.05.20240411.8cb0044`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(john): `""`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/gamy92vgx2ply9r9byzvrqm6ydix4l32-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by AkechiShiro** [comment]
Here is what I see on my side so far : 
```
nix-shell -p python312Packages.django
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'shell'
         whose name attribute is located at /nix/store/wdzp0hmh5lbykyvchd1wihszcyab78qv-nixos-23.11/nixos/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildInputs' of derivation 'shell'

         at /nix/store/wdzp0hmh5lbykyvchd1wihszcyab78qv-nixos-23.11/nixos/pkgs/stdenv/generic/make-derivation.nix:395:7:

          394|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          395|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          396|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: nose3-1.3.8 not supported for interpreter python3.12
```

You should wait until nose support the python3.12 interpreter.

**Comment by vcunat** [comment]
This doesn't mention explicitly, but the issue only exists in `master..staging-next` currently (PR #303527).  On Hydra e.g. https://hydra.nixos.org/build/256380688

The regressed tests happen on *-linux, both pythons, django_4 only.

**Comment by AkechiShiro** [comment]
I was looking upstream but it may not be related to this issue then : https://github.com/jayvdb/nose3/issues/5 still I'll mention it, apparently a fork for the nose lib is better maintained upstream and compatible with python3.12.

https://github.com/mdmintz/pynose

There has been test failures reported as well on Python 3.11 upstream : https://github.com/jayvdb/nose3/pull/4

**Comment by mweinelt** [comment]
Fixed in df533183f7d435d668e72597889eaf51bbd4eed6
